api/api_public_blockchain.go

```diff
@@ -253,12 +253,14 @@ func (s *PublicBlockChainAPI) doCall(ctx context.Context, args CallArgs, blockNr
 		}
 	}
 	// Set default gas & gas price if none were set
-	gas, gasPrice := uint64(args.Gas), args.GasPrice.ToInt()
+	gas := uint64(args.Gas)
 	if gas == 0 {
 		gas = math.MaxUint64 / 2
 	}
-	if gasPrice.Sign() == 0 {
-		gasPrice = new(big.Int).SetUint64(defaultGasPrice)
+
+	gasPrice := new(big.Int).SetUint64(defaultGasPrice)
+	if args.GasPrice != nil {
```

Comment:
Before this, we used default gas price if gasPrice was zero. It's going to change the meaning of the API. Wouldn't that be a problem?

Reply:
Before this PR the given gas Price zero means 2 cases like below.
- user gave zero gas price.
- user didn't set the gas price.

But almost client usually doesn't set the gasPrice for `Call` API. Because gasPrice is not important for `Call`.

And because of defaultGasPrice, `state.SetBalance(msg.ValidatedSender(), math.MaxBig256)` code is needed. This made weird balance.

Reply:
If gas price is not meaningful for the `Call`, why do we have to keep the value set by the user? How about simply use `zero` always?

Reply:
Becase `gasPrice` is one parameter for EVM. 
So if user want to set it, we need to give a way to use their `gasPrice`.
We don't need to simplify the optional feature.